app/controllers/Leagues.java

```diff
@@ -13,214 +11,213 @@
-      
-      //return ok(LeagueList.render("League Team", LeagueDB.getLeague(leagueForm.get().id), leagueForm, listTeam, Secured.isLoggedIn(ctx())));
-      return redirect("/leagues/view/" + form.id);
-    }
-  }
-  
-  public static Result showLeague(Long id) {
-    
-    League league = LeagueDB.getLeague(id);
-    List<Team> listTeam = TeamDB.getTeams();
-    LeagueForm leagueForm = new LeagueForm();
-    Form<LeagueForm> emptyForm = Form.form(LeagueForm.class).fill(leagueForm);
-    
-    return ok(LeagueList.render("League Team", league, emptyForm, listTeam, Secured.isLoggedIn(ctx())));
-  }
-  
-  @Security.Authenticated(Secured.class)
-  public static Result allLeagues(String sort, Integer page) {
-
-    SearchFormData st = new SearchFormData();
-    Form<SearchFormData> stuff = Form.form(SearchFormData.class).fill(st);
-
-    Page<League> currPage = League.find(sort, page);
-    return ok(AllLeagues.render("All leagues", currPage, sort, stuff, Secured.isLoggedIn(ctx())));
-  }
+	/**
+	 * returns the LeaguesPage
+	 * 
+	 * @return the LeaguesPage
+	 */
+	@Security.Authenticated(Secured.class)
+	public static Result leagues(Long id) {
+		System.out.println("leagues id " + id);
+		League league = LeagueDB.getLeague(id);
+		List<Team> listTeam = league.getTeams();
+		LeagueForm leagueForm = new LeagueForm();
+		if(id != 1){
+			leagueForm = new LeagueForm(league);
+		}
+		Form<LeagueForm> emptyForm = Form.form(LeagueForm.class).fill(leagueForm);
+
+		return ok(LeagueList.render("Leagues", league, emptyForm, listTeam, Secured.isLoggedIn(ctx())));
+	}
+
+	@Security.Authenticated(Secured.class)
+	public static Result searchLeagues(Integer page) {
+		SearchFormData st = new SearchFormData();
+		Form<SearchFormData> stuff = Form.form(SearchFormData.class).fill(st);
+
+		Form<SearchFormData> searcher = Form.form(SearchFormData.class).bindFromRequest();
+		SearchFormData st2 = searcher.get();
+
+		Page<League> currPage = League.find(st2.term, "leagueName asc", page);
+
+		return ok(SearchLeagues.render("Leagues", currPage, Secured.isLoggedIn(ctx()), st2.term));
+	}
+
+	public static Result editSchedule(){
+		LeagueForm leagueForm = new LeagueForm();
+		Form<LeagueForm> emptyForm = Form.form(LeagueForm.class).fill(leagueForm);
+		return ok(LeagueSchedule.render("Create Schedule", emptyForm, Secured.isLoggedIn(ctx())));
+	}
+
+	public static void generateSchedule(League league){
+
+		List<Team> opp = league.getTeams();
+		List<Team> home = new ArrayList<Team>();
+		List<Team> away = new ArrayList<Team>();
+
+		if(opp == null || opp.size() == 1){
+
+		}
+		else if(opp.size() < 3){
+			for(int i = 0; i < league.getDateList().size(); i++){
+				List<String> dates = league.getDateList();
+				league.addOpponent(dates.get(i), opp.get(0).getTeamName(), opp.get(1).getTeamName());
+			}
+		}
+		else{
+			for(int i = 0, j = opp.size() / 2; i < opp.size() / 2; i++, j++){
+				home.add(opp.get(i));
+				away.add(opp.get(j));
+				if((i+1)*2+1 == opp.size()){
```

Comment:
Oh, I remember this now.  It was the schedule generating algorithm.  It's pretty difficult to generate a schedule when you don't know the size of the league, or the number of games played